crates/sats/src/relation.rs

```diff
@@ -255,34 +384,77 @@ impl Header {
         )
     }
 
-    pub fn find_by_name(&self, field_name: &str) -> Option<&Column> {
-        self.fields.iter().find(|x| x.field.field_name() == Some(field_name))
+    /// Returns a slice of columns in the `Header`.
+    pub fn fields(&self) -> &[Column] {
+        &self.fields.columns
+    }
+
+    /// Returns the [Column] at the specified position.
+    pub fn column_by_pos(&self, pos: usize) -> &Column {
+        &self.fields[pos]
     }
 
-    pub fn column_pos<'a>(&'a self, col: &'a FieldName) -> Option<ColId> {
+    /// Returns the [FieldName] at the specified position.
+    pub fn field_by_pos(&self, pos: usize) -> Option<&FieldName> {
+        self.fields.column_by_pos(pos).map(|x| x.field.as_ref())
+    }
+
+    /// Returns the [Column] with the specified [FieldName].
+    pub fn column(&self, col: &FieldName) -> Option<&Column> {
         match col {
-            FieldName::Name { .. } => self.fields.iter().position(|f| &f.field == col),
-            FieldName::Pos { field, .. } => self
-                .fields
-                .iter()
-                .enumerate()
-                .position(|(pos, f)| &f.field == col || *field == pos),
+            FieldName::Name { .. } => self.fields.get_field(col),
+            FieldName::Pos { field, .. } => self.fields.column_by_pos(*field),
         }
-        .map(Into::into)
     }
 
-    pub fn column_pos_or_err<'a>(&'a self, col: &'a FieldName) -> Result<ColId, RelationError> {
-        self.column_pos(col)
+    /// Returns the [Column] with the specified `field_name`.
```

Comment:
Do we need all of these extra methods on Header? I was assuming we would just update the `find_pos_by_name` method.

Reply:
Is for streamlining the calls elsewhere...